src/app/public/modules/modal/modal.component.ts

```diff
@@ -164,8 +192,25 @@ export class SkyModalComponent implements AfterViewInit {
     this.hostService.onOpenHelp(this.helpKey);
   }
 
-  public closeButtonClick() {
-    this.hostService.onClose();
+  public close() {
+    if (!this.closeConfirmation) {
+      this.hostService.onClose();
+    } else {
+      const confirmConfig = isBoolean(this.closeConfirmation) ? {} : this.closeConfirmation;
+      const modalInstance: SkyModalInstance = this.modalService.open(
```

Comment:
Curious why we can't use the `SkyConfirmService` for this, instead of creating a new component?
https://github.com/blackbaud/skyux-modals/blob/master/src/app/public/modules/confirm/confirm.service.ts#L24

Reply:
I was attempting to use that before, but forgot that it would be a circular dependency between the `ConfirmModule` and `ModalModule`